Opentrons/opentrons
# feat(api): Add locks on direct data access for mag and temp module

## overview

Add locks on both the temperature and magnetic modules to prevent multi-access to data from the hardware.

## review requests

**Test Plan**
1. Ensure you can still set/deactivate temperature in the app
2. Run an API v1 protocol with a tempdeck and set temp low (maybe 5C) so that you can cancel before the temperature reaches the target. You should still be able to deactivate the temperature module after in the app, and then set temp again in the app.
3. Repeat #2 with an API v2 protocol.
4. With a magnetic module, ensure that you can plug/unplug and have the robot continue to be able to recognize it.
5. Use a magnetic module in a v1 and v2 protocol
